### PR TITLE
schedule: add IsFeatureSupported in Cluster Interface

### PR DESCRIPTION
**Pull request by HunDunDM**
Signed-off-by: Zheng Xiangsheng <hundundm@gmail.com>

### What problem does this PR solve?

* schedule: add `IsFeatureSupported` in `Cluster` Interface
* add `JointConsensus ` Feature type

### What is changed and how it works?

* mv `version.go` to `versioninfo` package

### Check List

Tests

- Unit test

### CODE REVIEW

File: server/cluster/cluster_worker.go [hunk 1]
```diff
@@ -16,15 +16,17 @@ package cluster
 import (
 	"bytes"
 
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/cluster/cluster_worker.go:21]: rm empty line

File: server/grpc_service.go [hunk 2]
```diff
@@ -25,12 +25,14 @@ import (
 	"github.com/pingcap/kvproto/pkg/metapb"
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/log"
-	"github.com/pingcap/pd/v4/server/cluster"
-	"github.com/pingcap/pd/v4/server/core"
 	"github.com/pkg/errors"
 	"go.uber.org/zap"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
+
```
Comment (disksing) [inline server/grpc_service.go:32]: rm empty line

File: server/server.go [hunk 3]
```diff
@@ -36,6 +36,15 @@ import (
...
 	"github.com/pingcap/log"
+	"github.com/pingcap/sysutil"
+	"github.com/pkg/errors"
+	"github.com/urfave/negroni"
+	"go.etcd.io/etcd/clientv3"
+	"go.etcd.io/etcd/embed"
+	"go.etcd.io/etcd/pkg/types"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+
```
Comment (disksing) [inline server/server.go:47]: rm empty line

File: server/util.go [hunk 4]
```diff
@@ -22,14 +22,14 @@ import (
 
 	"github.com/pingcap/kvproto/pkg/pdpb"
 	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	"go.etcd.io/etcd/clientv3"
+	"go.uber.org/zap"
+
```
Comment (disksing) [inline server/util.go:28]: rm empty line

**Comment by disksing** [comment]
I think `IsFeatureSupported` and `IsCompatible` should merge into one function.
Consider this case: A feature is introduced by 5.1.1, after all tikv upgrade to 5.1.1, the feature is enabled. Then another 5.1.0 tikv starts to join the cluster. Since PD use `IsCompatible` to check, the TiKV will join successfully.

File: server/cluster/cluster.go [hunk 5]
```diff
@@ -1329,13 +1326,18 @@ func (c *RaftCluster) changedRegionNotifier() <-chan *core.RegionInfo {
...
 // IsFeatureSupported checks if the feature is supported by current cluster.
-func (c *RaftCluster) IsFeatureSupported(f Feature) bool {
+func (c *RaftCluster) IsFeatureSupported(f versioninfo.Feature) bool {
 	c.RLock()
 	defer c.RUnlock()
 	clusterVersion := *c.opt.GetClusterVersion()
-	minSupportVersion := *MinSupportedVersion(f)
-	return !clusterVersion.LessThan(minSupportVersion)
+	minSupportVersion := *versioninfo.MinSupportedVersion(f)
+	if versioninfo.IsCompatible(minSupportVersion, version4_0) {
```
Comment (disksing) [inline server/cluster/cluster.go:1337]: Need some comment here. the rest LGTM

File: server/cluster/cluster.go [hunk 6]
```diff
@@ -1329,13 +1326,21 @@ func (c *RaftCluster) changedRegionNotifier() <-chan *core.RegionInfo {
 	return c.changedRegions
 }
 
+var version4_0 = *versioninfo.MinSupportedVersion(versioninfo.Version4_0)
```
Comment (rleungx) [inline server/cluster/cluster.go:1329]: Why do we need a global variable?

**Pull request merged**